### DISCUSSION TITLE
Plotting two rects as one

### DISCUSSION THREAD

**Discussion by simsam8** [pyqtgraph/pyqtgraph#2912] [Q&A] [answered]
Hi,

I am making a backtesting program and I want to add the functionality to draw positions. The basic idea is to create a rect which contains two individual rects, like the example images below (from TradingView).

Currently I am only able to plot to separate rects, which are not linked in any way. (bottom image)

Any thoughts on how to implement this would be much appreciated.

![image](https://github.com/pyqtgraph/pyqtgraph/assets/80512369/0ba42448-71a1-4477-8f7f-2cf5367b1ee9)
![image](https://github.com/pyqtgraph/pyqtgraph/assets/80512369/d6f17f33-a265-4e48-96fa-10fb73d99338)

![image](https://github.com/pyqtgraph/pyqtgraph/assets/80512369/251955c1-ac97-4b7f-b94b-a92aa2cc201e)

**Comment by ntjess** [comment]
How would you like them linked? Just that the positions are aligned on the green bottom edge and red top edge? Or do they have to move together/etc.?

**Reply by simsam8** [reply, marked as answer]
I figured out a solution. Instead of creating two rectangles/ROIs, I instead extended the ROI class. It takes a given ratio, and depending on what type of position it is, it paints the tp and sl accordingly.

```python
class FinPosition(pg.ROI):
    def __init__(self, ax, brush_tp, position_type, brush_sl, ratio=.5, *args, **kwargs):
        self.ax = ax
        self.brush_tp = brush_tp
        self.brush_sl = brush_sl
        self.position_type = position_type

        if position_type == "long":
            ratio=1-ratio
        self.ratio = ratio

        super().__init__(*args, **kwargs)

        center = [0.5, ratio]

        dy = -self.size()[1]*self.ratio
        self.translate(0, dy, False)
            
        if self.resizable:
            self.addScaleHandle([1, 0.5], [0, 0.5])
            self.addScaleHandle([0, 1], [0, center[1]])
            self.addScaleHandle([0, 0], [0, center[1]])
        
    def paint(self, p, *args):
        r = QtCore.QRectF(0, 0, self.state['size'][0], self.state['size'][1]).normalized()
        if self.position_type == "long":
            first_brush = self.brush_sl
            second_brush = self.brush_tp
        elif self.position_type == "short":
            first_brush = self.brush_tp
            second_brush = self.brush_sl
        else:
            raise Exception(f"{self.position_type}: is not a valid position type. Must be 'long' or 'short'")
            
        p.setPen(self.currentPen)
        p.setBrush(first_brush)

        p.translate(r.left(), r.top())
        p.scale(r.width(), r.height())

        p.scale(1, self.ratio)
        p.drawRect(0, 0, 1, 1)

        p.drawLine(0,1,1,1)

        p.setBrush(second_brush)
        p.translate(0, 1)
        p.scale(1,(1-self.ratio)/self.ratio)

        p.drawRect(0, 0, 1, 1)

    def addScaleHandle(self, *args, **kwargs):
        if self.resizable:
            self.handleSize = 15
            self.handlePen = pg.mkPen("blue")
            super().addScaleHandle(*args, **kwargs)
```

![image](https://github.com/pyqtgraph/pyqtgraph/assets/80512369/1e3dfddc-e71f-42ad-ad26-27e5ec243ee9)
